pkg/apis/pipeline/v1alpha1/taskrun_types.go

```diff
@@ -112,6 +126,11 @@ type TaskRunStatus struct {
 	// +optional
 	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
 
+	// ExpirationTime is the time the build expired.
+	// If TaskRun is built from PipelineRun, the time is the PipelineRun's ExpirationTime.
+	// If TaskRun doesn't have PipelineRun OwnerReference, then the time is TaskRun's ExpirationTime
```

Comment:
NIT: I'm not sure this comment belongs here, since it's the controller enforcing that, so this should sit next to the corresponding logic to avoid the comment becoming stale.